app/services/next_course_creation_step_service.rb

```diff
@@ -1,48 +1,64 @@
 class NextCourseCreationStepService
-  def execute(current_step:, current_provider:)
-    case current_step
-    when :level
-      :subjects
-    when :subjects
-      :age_range
-    when :age_range
-      :outcome
-    when :outcome
-      handle_outcome(current_provider)
-    when :fee_or_salary
-      :full_or_part_time
-    when :apprenticeship
-      :full_or_part_time
-    when :full_or_part_time
-      :location
-    when :location
-      handle_location(current_provider)
-    when :accredited_body
-      :entry_requirements
-    when :entry_requirements
-      :applications_open
-    when :applications_open
-      :start_date
-    when :start_date
-      :confirmation
-    end
+  def execute(current_step:, current_provider:, level:)
+    get_next_step(get_workflow(current_provider, level), current_step)
```

Comment:
I find this a little bit hard to read, and unfortunately I don't think pushing it into separate methods helps. How about:

```ruby
  def execute(current_step_name:, current_provider:, level:)
    steps = get_workflow(current_provider, level)
    current_step = steps.find_index(current_step_name)
    steps[current_step.next]
  end
```

You might even consider renaming `get_workflow` to `get_workflow_steps`.

Reply:
How is it now? I still think pushing it into separate methods helps a bit so I kept that but got rid of the nesting and renamed `get_workflow`